test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ESClientYamlSuiteTestCase.java

```diff
@@ -306,14 +306,31 @@ public void test() throws IOException {
 
         try {
             for (ExecutableSection executableSection : testCandidate.getTestSection().getExecutableSections()) {
-                executableSection.execute(restTestExecutionContext);
+                executeSection(executableSection);
             }
         } finally {
             logger.debug("start teardown test [{}]", testCandidate.getTestPath());
             for (DoSection doSection : testCandidate.getTeardownSection().getDoSections()) {
-                doSection.execute(restTestExecutionContext);
+                executeSection(doSection);
             }
             logger.debug("end teardown test [{}]", testCandidate.getTestPath());
         }
     }
+
+    /**
+     * Execute an {@link ExecutableSection}, careful to log its place of origin on failure.
+     */
+    private void executeSection(ExecutableSection executableSection) {
+        try {
+            executableSection.execute(restTestExecutionContext);
+        } catch (Exception e) {
+            throw new RuntimeException(errorMessage(executableSection, e), e);
+        } catch (AssertionError e) {
+            throw new AssertionError(errorMessage(executableSection, e), e);
+        }
+    }
+
+    private String errorMessage(ExecutableSection executableSection, Throwable t) {
+        return "Failure at [" + testCandidate.getSuitePath() + ":" + executableSection.getLocation().lineNumber + "]: " + t.getMessage();
```

Comment:
maybe add which type of section it was, so that it's even safer to get rid of the double exception? That is the only useful bit I find in the original stacktrace that would otherwise get lost.